tests/unit/test_links.py

```diff
@@ -0,0 +1,65 @@
+# THIS FILE IS PART OF THE CYLC WORKFLOW ENGINE.
+# Copyright (C) NIWA & British Crown (Met Office) & Contributors.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""Check links inserted into internal documentation.
+
+Reason for doing this here:
+- Some links don't appear to be being picked up by Cylc-doc linkcheck.
+- As we have more links it's worth checking them here, rather than waiting
+  for them to show up in Cylc.
+"""
+
+from pathlib import Path
+import re
+from shlex import split
+from subprocess import run
+import pytest
+import urllib
+
+EXCLUDE = [
+    'http://www.gnu.org/licenses/',
+    'http://my-site.com/workflows/%(workflow)s/index.html',
+    'http://ahost/%(owner)s/notes/%(workflow)s',
+    'http://my-site.com/workflows/%(workflow)s/'
+]
+
+def get_links():
+    searchdir = Path(__file__).parent.parent.parent / 'cylc/flow'
+    for file_ in searchdir.rglob('*.py'):
+        for url in re.findall(r'(https?:\/\/.*?)[\n\s\>`"\',]', file_.read_text()):
+            if url not in EXCLUDE:
+                yield {'file': file_, 'url': url}
```

Comment:
It may be worth return a set to avoid duplication. Also I wonder if it's worth trying to skip comments (if there is a `#` on the line before the "http")?

Reply:
We don't want to skip links in comments, because these are just as valid for future devs.

Reply:
Sorry, and a secondary thought - in some cases links in commented lines are ingested by Cylc CLI and become part of the help docs.

Reply:
Hmm, that sounds weird. Can you give an example?

Reply:
```bash
$ cylc play --help
...
  --initial-cycle-point=CYCLE_POINT or OFFSET, --icp=CYCLE_POINT or OFFSET
                        Set the initial cycle point. Required if not defined
                        in flow.cylc. May be either an absolute point or an
                        offset: See https://bit.ly/3MYHqVh
...

# vs
$ cylc play --help
...
  --initial-cycle-point=CYCLE_POINT or OFFSET, --icp=CYCLE_POINT or OFFSET
                        Set the initial cycle point. Required if not defined
                        in flow.cylc. May be either an absolute point or an
                        offset: See https://cylc.github.io/cylc-doc/latest/htm
                        l/reference/config/workflow.html#flow.cylc[scheduling]
                        initial cycle point
...
```

Reply:
I meant an example of a commented line being included in the docs.

Reply:
> I meant an example of a commented line being included in the docs.

No, I can't. I was wrong about this. It's a holdover in my head from Cylc 7.I think the point about the CLI stands though.